### PR TITLE
net: rpc: openthread: dynamic allocation for CLI commands

### PR DESCRIPTION
**Pull request by konradderda**

### CODE REVIEW

**Review by bpienk — APPROVED** [review]

**Review by canisLupus1313 — APPROVED** [review]

**Review by lmaciejonczyk — APPROVED** [review]

File: include/nrf_rpc/nrf_rpc_serialize.h [hunk 1]
```diff
@@ -232,7 +232,7 @@ char *nrf_rpc_decode_str(struct nrf_rpc_cbor_ctx *ctx, char *buffer, size_t buff
 /** @brief Decode a string pointer and length. Moves CBOR pointer past string on success.
  *
  * @param[in,out] ctx CBOR decoding context.
- * @param[out]  size String length.
+ * @param[out] len String length.
```
[the remark this answers is not in the document]
Comment (konradderda) [reply include/nrf_rpc/nrf_rpc_serialize.h:235]: Fixed

File: subsys/net/openthread/rpc/server/ot_rpc_cli.c [hunk 2]
```diff
@@ -68,32 +68,43 @@ NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_cli_init, OT_RPC_CMD_CLI_INIT, ot_
...
+		buffer = malloc(len + 1);
+		if (buffer) {
+			memcpy(buffer, ptr, len);
+			buffer[len] = '\0';
+		}
+	}
 
 	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
 		ot_rpc_report_cmd_decoding_error(OT_RPC_CMD_CLI_INPUT_LINE);
 		return;
```
[the remark this answers is not in the document]
Comment (konradderda) [reply subsys/net/openthread/rpc/server/ot_rpc_cli.c]: Good catch, fixed

File: subsys/net/openthread/rpc/server/ot_rpc_cli.c [hunk 3]
```diff
@@ -68,32 +68,45 @@ NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_cli_init, OT_RPC_CMD_CLI_INIT, ot_
...
+
+	if (ptr) {
+		buffer = malloc(len + 1);
+		if (buffer) {
+			memcpy(buffer, ptr, len);
+			buffer[len] = '\0';
+		}
+	}
 
 	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
```
Comment (anangl) [inline subsys/net/openthread/rpc/server/ot_rpc_cli.c:87]: Why not move this before the allocation and copying which turn out to be unnecessary when this fails? This extra call to `free` would not be needed then.

**Review by anangl — APPROVED** [review]

File: subsys/net/openthread/rpc/server/ot_rpc_cli.c (hunk 3 above)
Comment (konradderda) [reply subsys/net/openthread/rpc/server/ot_rpc_cli.c:87]: `nrf_rpc_decoding_done_and_check()` frees the buffer which holds the data to be stored in the `buffer`.

**Review by pdunaj — APPROVED** [review]
checked nrf_rpc commit

**Pull request merged**